### DISCUSSION TITLE
[FR] Custom asset transformers

### DISCUSSION THREAD

**Discussion by leevigraham** [craftcms/cms#11225] [Ideas]
It would be great if we could register a custom asset transformer to replace the default craft transformer. The new transformer could create urls for third-party services e.g.: imgix, cloudinary, cloudflare images. 

Where this could get tricky is exposing all the options of a transformer… for example: imgix has lots and lots of options on top of the standard crop, height, width.

Maybe the image transformer could be implemented as an application component like db, cache etc. https://www.yiiframework.com/doc/guide/2.0/en/structure-application-components

It looks like there is some code related to this idea:

<img width="604" alt="Screen Shot 2022-05-15 at 5 59 08 pm" src="https://user-images.githubusercontent.com/25124/168467321-b6e5272e-9bab-4522-9a25-44c4af8b89bf.png">

<img width="590" alt="Screen Shot 2022-05-15 at 8 00 28 pm" src="https://user-images.githubusercontent.com/25124/168467275-4bb0c372-7bc3-4c20-8831-2dc7b3e19533.png">

**Comment by thupsi** [comment]
Hi! Take a look at commit https://github.com/craftcms/cms/pull/10367

**Reply by leevigraham** [reply]
That PR mentions custom image transformers but I can't find any implementation detail. There's only one hook that is used when assets are deleted.

**Reply by brandonkelly** [reply]
We’re working to improve the API and will publish more details about it once it’s in a state we’re happy with.

**Reply by leevigraham** [reply]
Let me know if you want a tester  / bug finder!

**Comment by leevigraham** [comment]
<img width="770" alt="Screen Shot 2022-05-17 at 1 27 29 am" src="https://user-images.githubusercontent.com/25124/168628511-a2b6ba6a-fcf8-4b72-aeb8-e04cd9cca471.png">
